dbt_platform_helper/providers/ecs.py

```diff
@@ -45,19 +89,32 @@ def get_or_create_task_name(self, addon_name: str, parameter_name: str) -> str:
             random_id = "".join(random.choices(string.ascii_lowercase + string.digits, k=12))
             return f"conduit-{self.application_name}-{self.env}-{addon_name}-{random_id}"
 
-    def get_ecs_task_arns(self, cluster_arn: str, task_name: str):
+    def get_ecs_task_arns(self, cluster_arn: str, task_def_family: str):
         """Gets the ECS task ARNs for a given task name and cluster ARN."""
         tasks = self.ecs_client.list_tasks(
             cluster=cluster_arn,
             desiredStatus="RUNNING",
-            family=f"copilot-{task_name}",
+            family=task_def_family,
         )
 
         if not tasks["taskArns"]:
             return []
 
         return tasks["taskArns"]
 
+    def exec_task(self, cluster_arn: str, task_arn: str):
```

Comment:
If you pass in "subprocess.call" as a "call_fn" parameter or similar then you can put some tests around this retry functionality.

Reply:
This is something I feel like I hit a lot and starts to feel like it isn't worth the effort but maybe that is a blind spot for myself.
I like injecting it as a dependency to test for sure but testing retries feels difficult. As we would want to mock then add a side effect for `subprocess` which would then succeed on call count 2/3. The difficulty for myself is tracking the call count.
Or would you expectation of a test be to let the number of retries exceed as a valid test?